src/cdn_definitions/_impl/__init__.py

```diff
@@ -1,26 +1,78 @@
 import json
 import os
+import warnings
+
+import requests
 import yaml
 
 
-def load_data():
-    # Load data from first existing of these
-    candidate_paths = [
-        os.path.join(os.path.dirname(os.path.dirname(__file__)), "data.yaml"),
-        "/usr/share/cdn-definitions/data.yaml",
-    ]
+def get_data(data_paths):
+    """Loads data from YAML or JSON format into a dictionary.
+
+    Args:
+        data_paths (list[str]): A list of local paths or URLs to a JSON or YAML data file.
+
+    Returns:
+        dict: The data from the local path or URL in the form of a dictionary.
+
+    Raises:
+        RuntimeError: If all attempted data paths are invalid, a RuntimeError will be raised.
+
+    """
+    for p in data_paths:
+        _, ext = os.path.splitext(p)
+
+        if p.startswith("http"):
+            if ext.lower() == ".json":
+                return json.loads(requests.get(p).json())
```

Comment:
I think the method of doing requests needs to be beefed up a little:

- We've got to check status codes on the responses.  If we just try to unconditionally parse them as JSON or YAML then we will throw away error info when the requests fail (see RHELDST-4813 for an example where a tool just tries to parse server response as JSON without checking the status code)
- Could we please apply some modest retries here so we don't immediately fail if the request doesn't work on the first try?  In most cases where we don't add that, we have to come back and add it later.  In this case it would look similar to https://findwork.dev/blog/advanced-usage-python-requests-timeouts-retries-hooks/#retry-on-failure .